app/src/main/java/com/kylecombes/micrepeater/ScreenSlidePageFragment.java

```diff
@@ -26,22 +27,48 @@ public static ScreenSlidePageFragment newInstance(Integer counter) {
         fragment.setArguments(args);
         return fragment;
     }
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         if (getArguments() != null) {
             counter = getArguments().getInt(ARG_COUNT);
         }
+
     }
+
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         return inflater.inflate(layout_files[counter], container, false);
     }
+
     @Override
-    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+    public void onViewCreated(@NonNull final View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
+        if (counter == 4) {
+            Switch datareporting = view.findViewById(R.id.data_switch);
+            // By default, set reportdata to true
+            SharedPreferences preferences = getActivity().getSharedPreferences("PREFERENCE", Context.MODE_PRIVATE);
+            if (preferences.contains("reportdata")) {
```

Comment:
I think we can be a bit more precise here. How about something like `useFirebase` or `haveAnalyticsConsent`?

Reply:
Fixed